### DISCUSSION TITLE
how to use Web3 with this new version?

### DISCUSSION THREAD

**Discussion by quantix-nft** [WalletConnect/walletconnect-monorepo#3194] [Developer Support]
Hello,
just started migrating from v1. All this documentation is just blowing me and I think I won't be able to keep my dapp working.
In v1 I used to get provider object after the modal.connect promise. Then I used to pass that to new Web3 object as constructor.
All my dapp uses provider object and web3 to work, and I cannot change everything now. I would have only wanted to change WalletConnect with the new version not everything :(
Please help me

**Comment by glitch-txs** [comment]
You can integrate web3modal with [wagmi's ethers adapter](https://wagmi.sh/react/ethers-adapters) and change ethers for web3. A better solution is web3modal v3 but it's still under development.

**Reply by QianDuoDuo-web** [reply]
Is there any documentation for v2, or a code snippet?

**Reply by QianDuoDuo-web** [reply]
web3modal integrates with wagmi's ethers adapter and changes web3's ethers
